### PR TITLE
Show renewal pricing for domain registrations.

### PR DESCRIPTION
**Pull request by delputnam**
#### Changes proposed in this Pull Request

In order to make it clearer that domain registration that is included in the price of a plan doesn't include the annual renewal, we want to update the wording and show the renewal price.

#### Testing instructions

Search for a new domain in the following cases:
- Site with a plan and an unused domain credit
- Site without a plan
- Site with a blogger plan

Make sure that the wording is appropriate for each case.

Make sure that the price still shows correctly when adding a new domain to a site that has no domain credit.

Make sure that the free subdomain suggestions are still listed as free.

Check the wording in the shopping cart and make sure it specifies that only the first year is free.

### CODE REVIEW

File: client/components/domains/domain-product-price/index.jsx [hunk 1]
```diff
@@ -25,63 +25,56 @@ class DomainProductPrice extends React.Component {
...
+
+		return <div className="domain-product-price__free-text">{ message }</div>;
 	}
 
 	renderFreeWithPlanPrice() {
 		return (
-			<span className="domain-product-price__price">
-				{ this.props.translate( '%(cost)s {{small}}/year{{/small}}', {
+			<div className="domain-product-price__price">
+				{ this.props.translate( 'Renewal: %(cost)s {{small}}/year{{/small}}', {
```
Comment (a8ci18n) [inline client/components/domains/domain-product-price/index.jsx]: :disappointed: 36 existing translations will be lost with this change.
If the string must change, the following option has already been translated **1** time:
`translate( '{{small}}+%(cost)s / year{{/small}}' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/components/domains/domain-product-price/index.jsx [hunk 2]
```diff
@@ -25,63 +25,56 @@ class DomainProductPrice extends React.Component {
...
-				</span>
-			</div>
-		);
+	renderFreeWithPlanText() {
+		const { translate } = this.props;
+
+		let message;
+		switch ( this.props.rule ) {
+			case 'FREE_WITH_PLAN':
+				message = this.props.translate( 'First year free with your plan' );
```
Comment (klimeryk) [inline client/components/domains/domain-product-price/index.jsx]: `translate` ;)

**Comment by delputnam** [comment]
Thanks for the careful review, @klimeryk!

I think  I've fixed everything you noticed now.  Here are some screenshots showing what you should see:

#### Domain suggestions for a pre-DWPO site without a paid plan or a site with a paid plan, but without a free domain credit:
screen shot 2019-02-08 at 1 15 40 pm

#### Domain suggestions for a post-DWPO site without a paid plan:
screen shot 2019-02-08 at 1 07 48 pm

#### Domain suggestions for a site with a free domain credit:
screen shot 2019-02-08 at 1 11 34 pm

#### Domain suggestions for a site with a paid plan that requires an upgrade to purchase a domain:
screen shot 2019-02-08 at 1 13 31 pm

#### When mapping a domain on a pre-DWPO site without a paid plan:
screen shot 2019-02-08 at 10 24 04 am

#### When mapping a domain a post-DWPO site without a paid plan:
screen shot 2019-02-08 at 12 44 26 pm

#### When mapping a domain on a site with a paid plan:
screen shot 2019-02-08 at 10 24 43 am

#### A transfer on a pre-DWPO site without a paid plan or a site with a paid plan, but no domain credit:
screen shot 2019-02-08 at 12 48 13 pm

#### A transfer on a post-DWPO site without a paid plan:
Before the domain is entered:
screen shot 2019-02-08 at 12 06 05 pm

After the domain is entered:
screen shot 2019-02-08 at 12 06 13 pm

#### A transfer on site with a paid plan and a free domain credit:
Before the domain is entered:
screen shot 2019-02-08 at 12 09 35 pm

After the domain is entered:
screen shot 2019-02-08 at 12 09 42 pm

**Comment by delputnam** [comment]
Thanks for the final checks, @klimeryk!

I have better-aligned the price text and cost as well as normalized the size between the various breakpoints as you suggested.

screen shot 2019-02-14 at 5 33 36 pm

**Pull request merged**